From c619ddbb06b16434ffec4862c508318fd7980428 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 13 Jun 2016 17:23:15 +0300
Subject: [PATCH 497/538] Revert "fix: i2c: update buffer array type to be u8,
 instead of u32"

commit 1279471cb41d ("fix: i2c: update buffer array type to be u8,
instead of u32")
The previous patch fix messages with length 0/1 and not all
the messages size

Change-Id: I81e3b6d25121d7fd77d9141052fa8443efca20e0
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30431
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
---
 drivers/i2c/busses/i2c-mv64xxx.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/i2c/busses/i2c-mv64xxx.c b/drivers/i2c/busses/i2c-mv64xxx.c
index 7a8e818..ef7b35d 100644
--- a/drivers/i2c/busses/i2c-mv64xxx.c
+++ b/drivers/i2c/busses/i2c-mv64xxx.c
@@ -434,10 +434,10 @@ static void
 mv64xxx_i2c_read_offload_rx_data(struct mv64xxx_i2c_data *drv_data,
 				 struct i2c_msg *msg)
 {
-	u8 buf[2];
+	u32 buf[2];
 
-	buf[0] = readb(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_LO);
-	buf[1] = readb(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_HI);
+	buf[0] = readl(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_LO);
+	buf[1] = readl(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_HI);
 
 	memcpy(msg->buf, buf, msg->len);
 }
@@ -592,12 +592,12 @@ static void
 mv64xxx_i2c_prepare_tx(struct mv64xxx_i2c_data *drv_data)
 {
 	struct i2c_msg *msg = drv_data->msgs;
-	u8 buf[2];
+	u32 buf[2];
 
 	memcpy(buf, msg->buf, msg->len);
 
-	writeb(buf[0], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_LO);
-	writeb(buf[1], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_HI);
+	writel(buf[0], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_LO);
+	writel(buf[1], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_HI);
 }
 
 static int
-- 
1.9.1

